app/forms/common/Time.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.common
+
+import play.api.data.Forms.{optional, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+import utils.validators.forms.FieldValidator.isContainedIn
+
+case class Time(hour: Option[String], minute: Option[String]) {
+
+  def isEmpty: Boolean = hour.isEmpty && minute.isEmpty
+  def nonEmpty: Boolean = hour.nonEmpty && minute.nonEmpty
+}
+
+object Time {
+  implicit val format = Json.format[Time]
+
+  private val validHours = (0 to 23).toList.map(_.toString)
+  private val validMinutes = (0 to 59).toList.map(_.toString)
+
+  val mapping = Forms
+    .mapping(
+      "hour" -> optional(text().verifying("dateTime.time.hour.error", isContainedIn(validHours))),
```

Comment:
I would question the requirement of why should user enter hour and minute of a document produced? it is not really practical , if it is needed for backend data population we should be able to do define this in our service

Reply:
This class was created because it is similar to `Date`. Since I had to create one I decided to write another as well. I was also planning to use them for Movement journey, but they were moved to separate repository in the meantime.